packages/ra-core/src/dataProvider/useDelete.spec.tsx

```diff
@@ -273,16 +273,16 @@ describe('useDelete', () => {
             screen.getByText('Delete first post').click();
             await waitFor(() => {
                 expect(screen.queryByText('success')).toBeNull();
-                expect(screen.queryByText('Hello')).not.toBeNull();
-                expect(screen.queryByText('World')).not.toBeNull();
                 expect(screen.queryByText('mutating')).not.toBeNull();
             });
+            expect(screen.queryByText('Hello')).not.toBeNull();
+            expect(screen.queryByText('World')).not.toBeNull();
             await waitFor(() => {
```

Comment:
the default timeout for waitFor is 1000ms. The tested story adds a delay of 1000ms to the delete. It's logical that this may fail from time to time. 

I advise to reduce the delay to 500 in the story to make it more robust